### PR TITLE
Adds Brave server integration tests and fixes test failures

### PR DESCRIPTION
**Pull request by llinder**

### CODE REVIEW

File: akka-http/src/main/scala/smartthings/brave/scala/akka/http/extension/AkkaHttpTracingExtensionImpl.scala [hunk 1]
```diff
@@ -1,33 +1,40 @@
...
 }
 
-class AkkaHttpTracingExtensionImpl(config: Config) extends Extension {
+class AkkaHttpTracingExtensionImpl(config: Config, httpTracingSupplier: Supplier[Option[HttpTracing]]) extends Extension {
 
-  lazy val httpTracing: HttpTracing = {
+  lazy val httpTracing: HttpTracing = httpTracingSupplier.get().getOrElse {
 
-    val tracing = TracingConfigurator(config.getConfig("brave.tracing"))
+      println("lazy http tracing create from config")
```
Comment (janderson007) [inline akka-http/src/main/scala/smartthings/brave/scala/akka/http/extension/AkkaHttpTracingExtensionImpl.scala]: log.debug?

File: akka-http/src/main/scala/smartthings/brave/scala/akka/http/internal/AkkaHttpServerAdapter.scala [hunk 2]
```diff
@@ -1,17 +1,57 @@
...
+
+  // TODO is there a good way to return a template route for span names?
+  override def route(response: (HttpRequest, HttpResponse)): String = super.route(response)
+
+  override def path(request: HttpRequest): String = {
+    request.uri.path.toString()
+  }
+
+  override def url(request: HttpRequest): String = {
+    println(request.uri.toString())
```
Comment (janderson007) [inline akka-http/src/main/scala/smartthings/brave/scala/akka/http/internal/AkkaHttpServerAdapter.scala]: debug?

File: akka-http/src/main/scala/smartthings/brave/scala/akka/http/extension/AkkaHttpTracingExtensionImpl.scala (hunk 1 above)
Comment (llinder) [reply akka-http/src/main/scala/smartthings/brave/scala/akka/http/extension/AkkaHttpTracingExtensionImpl.scala]: oops.  will delete that. thanks

File: akka-http/src/main/scala/smartthings/brave/scala/akka/http/internal/AkkaHttpServerAdapter.scala (hunk 2 above)
Comment (llinder) [reply akka-http/src/main/scala/smartthings/brave/scala/akka/http/internal/AkkaHttpServerAdapter.scala]: I need to add style checks.  Will do that after the client integration tests pass.

**Pull request merged**